### PR TITLE
Logs UI

### LINKED ISSUE #124: Logs UI
Get public logs setup on a hosted machine.  (more details to be added)

### LINKED ISSUE #141: Putting data to beanstalk tube fails with JOB_TOO_BIG error
When scanner tries to put data to beanstalkd tube, it fails with JOB_TOO_BIG error:
```shell
Jan 02 01:48:31 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:31,244 - container-pipeline - INFO - Result file: /var/lib/atomic/pipeline-scanner/2017-01-02-01-48-18-568129/_sha256:e265750f2f5ff3ee61b929cddb37a2a74ba5127a6732e9138b87bf3536ed41c2/image_scan_results.json
Jan 02 01:48:31 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:31,246 - container-pipeline - INFO - Unmounting image's rootfs from /sha256:e265750f2f5ff3ee61b929cddb37a2a74ba5127a6732e9138b87bf3536ed41c2
Jan 02 01:48:33 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:33,005 - container-pipeline - INFO - Finished executing scanner: pipeline-scanner
Jan 02 01:48:33 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:33,006 - container-pipeline - INFO - Finished running registry.centos.org/pipeline-images/pipeline-scanner scanner.
Jan 02 01:48:33 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:33,007 - container-pipeline - INFO - Removing the image: c2bf.rdu2.centos.org:5000/sgk/mattermost-gitlab-integration:20161230071406
Jan 02 01:48:33 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:33,950 - container-pipeline - INFO - Finished executing all scanners.
Jan 02 01:48:33 cccp-atomic-scan.c2bg.rdu2.centos.org worker_start_scan.py[16901]: 2017-01-02 01:48:33,970 - container-pipeline - CRITICAL - ('put', 'JOB_TOO_BIG', [])
```

### PR DESCRIPTION
**Pull request by navidshaikh**
Fixes #124 
Fixes #141

### CODE REVIEW

**Comment by navidshaikh** [comment]
rebased on PR #130 , since there is dependent code in mentioned PR.

**Comment by navidshaikh** [comment]
@dharmit @bamachrn @rtnpro @mohammedzee1000 : please review the PR

**Comment by navidshaikh** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
Looks pretty good to me

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by mohammedzee1000** [comment]
#dotests

**Comment by dharmit** [comment]
@navidshaikh it looks to me like we're hard-coding the value `https://registry.centos.org/pipeline-logs` in the code. But when we add tests (which we need to for overall repo and not just this PR), things might fail. Instead, can we do something similar to what's been done [here in ansible provisioning](https://github.com/CentOS/container-pipeline-service/blob/master/provisions/roles/scanner/tasks/main.yml#L107-L111)?

WDYT @bamachrn @rtnpro @mohammedzee1000

**Comment by navidshaikh** [comment]
> Are we making/going to make required changes to CI and provisioning through this PR?
@dharmit 
Are you mentioning about NFS provisioning ? The CI changes should go via a different PR.

File: beanstalk_worker/worker_dockerfile_lint.py [hunk 1]
```diff
@@ -32,6 +35,37 @@ def write_dockerfile(dockerfile):
...
+    logger.log(
+            level=logging.INFO,
+            msg="Linter log file: %s" % logs_file_path
+            )
+    try:
+        fin = open(logs_file_path, "w")
+    except IOError as e:
+        logger.log(
+            level=logging.CRITICAL,
+            msg="Failed to write linter logs on NFS share.")
```
Comment (navidshaikh) [inline beanstalk_worker/worker_dockerfile_lint.py]: I think its the same thing, if the file can't be opened in write mode, we failed to write the logs.

**Comment by navidshaikh** [comment]
> @navidshaikh it looks to me like we're hard-coding the value https://registry.centos.org/pipeline-logs in the code. But when we add tests (which we need to for overall repo and not just this PR), things might fail.

When we add tests, we'll make sure we modify this, in a separate PR.

File: beanstalk_worker/worker_start_scan.py [hunk 2]
```diff
@@ -101,6 +102,35 @@ def run_a_scanner(self, scanner, image_under_test):
...
+        logger.log(
+                level=logging.INFO,
+                msg="Scanner=%s result log file:%s" % (scanner, logs_file_path)
+                )
+        try:
+            fin = open(logs_file_path, "w")
+        except IOError as e:
+            logger.log(
+                level=logging.CRITICAL,
+                msg="Failed to write scanner=%s logs on NFS share." % scanner)
```
Comment (mohammedzee1000) [inline beanstalk_worker/worker_start_scan.py]: Hmm yea makes sense. What are we doing with an open file handle here.

Comment (navidshaikh) [inline beanstalk_worker/worker_start_scan.py]: I don't understand why are we categorizing the two errors, if (at the first place) a file can not be opened for writing, we failed writing to file, opening a file in write mode and failing to do so, is *fail to write*. IOError encapsulate that.

Comment (dharmit) [inline beanstalk_worker/worker_start_scan.py]: @navidshaikh let me give you a scenario.

- we're using an NFS share where we write data
- we open a file to write data to it
- connection to NFS share got broken before we could actually write to the file

What would happen in this situation? Are we going to be able to write to the file? 

Having `json.dump` in the `else` block is going to crash this program if I understand correctly.

So, in this situation, you could open the file for writing but couldn't write to it. Two different things.

@kbsingh @arrfab @bamachrn thoughts on the scenario I mentioned and its consequence? Please correct me if I'm wrong.

**Comment by dharmit** [comment]
@navidshaikh 
> Are you mentioning about NFS provisioning ? The CI changes should go via a different PR.

Okay, thanks.
> When we add tests, we'll make sure we modify this, in a separate PR.

Okay. Just to be clear, we need to modify it in the actual code and also the tests need to be written keeping the same in mind. Hard-coding FQDN will break things.

LGTM otherwise.

File: beanstalk_worker/worker_start_scan.py (hunk 2 above)
Comment (mohammedzee1000) [inline beanstalk_worker/worker_start_scan.py]: I personally feel all file write ops should be enclosed in try catch

Comment (navidshaikh) [inline beanstalk_worker/worker_start_scan.py]: updated as per suggestion.

**Comment by navidshaikh** [comment]
This is ready to merge. Verified the PR with openshift v1.2.1 and v1.3.1

Cc: @bamachrn

**Pull request merged**